### ISSUE TITLE
A11y: keyboard navigation in Tabs is counterintuitive

### ISSUE THREAD

**Issue by janett-baresel** [Esri/calcite-components#2124]
### Summary
When navigating through tabs via keyboard it is cumbersome to get from the tab to its content. 

Additionally, both sighted and non-sighted users (screen reader users) are unable to navigate tab content. Screen readers are not provided with any context residing in the `calcite-tab` component.

### Actual Behavior / Reproduction Steps
1. tab through tabs via keyboard - navigation goes from one to the next tab 👍 
2. press enter on one of the tabs - tab gets activated 👍 
3. tab again and the focus jumps to the next tab. Shift tab and the focus jumps to the previous tab. 
4. Shift tab all the way to the very first tab. Shift tab again and the focus jumps to the first element inside the tab container. 👎 

### Expected Behavior
1. tab through tabs via keyboard - navigation goes from one to the next tab
2. press enter on one of the tabs - tab gets activated 
3. tab again and the focus jumps into the tab container to the first element.

### Reproduction Sample
https://developers.arcgis.com/calcite-design-system/components/tabs

### Reproduction Version
beta.83

### Relevant Info
- [x] JAWS and Chrome/Firefox
- [x] NVDA and Firefox
- [x] Mac OS and VoiceOver

### Working W3C Example/Tutorial 
https://www.w3.org/WAI/ARIA/apg/example-index/tabs/tabs-manual.html

#### Code sample
```html
<div class="tabs">
  <div role="tablist" aria-labelledby="tablist-1" class="automatic">
    <button id="tab-1" type="button" role="tab" aria-selected="true" aria-controls="tabpanel-1">
      <span class="focus">Maria Ahlefeldt</span>
    </button>
    <button id="tab-2" type="button" role="tab" aria-selected="false" aria-controls="tabpanel-2" tabindex="-1">
      <span class="focus">Carl Andersen</span>
    </button>
  </div>

  <div id="tabpanel-1" role="tabpanel" tabindex="0" aria-labelledby="tab-1">
    <p>Maria Theresia Ahlefeldt (16 January 1755 – 20 December 1810) was a Danish, (originally German), composer.</p>
  </div>
  <div id="tabpanel-2" role="tabpanel" tabindex="0" aria-labelledby="tab-2" class="is-hidden">
    <p>Carl Joachim Andersen (29 April 1847 – 7 May 1909) was a Danish flutist, conductor and composer born in Copenhagen, son of the flutist Christian Joachim Andersen.</p>
  </div>
</div>
        
```

**Comment by macandcheese** [comment]
Hi @janett-baresel  - is there a direct link to a viewable example of this? I have tried to replicate it but have been unable to replicate step 4 above:

https://jsbin.com/jawibomese/edit?html,output

Re: the "expected behavior" - it sounds like the desire is for any subsequent "tab button press" to immediately focus available content inside the tab, as opposed to now where you need to "tab your way through" all the tabs before reaching the content of the active one?

FWIW, we allow navigation through a set of tabs with both tab / shift tab and the arrow keys. 

It seems like Google Material and IBM Carbon use "tab press" more in the way you have described, and navigation between tab items is only via key: 

https://material.io/components/tabs
https://www.carbondesignsystem.com/components/tabs/code/

(Seems like "home" / "end" support would also be nice for that)

**Comment by macandcheese** [comment]
Hi @janett-baresel - is there a direct link to a viewable example of this? I have tried to replicate it but have been unable to replicate step 4 above:

https://jsbin.com/jawibomese/edit?html,output

Re: the "expected behavior" - it sounds like the desire is for any subsequent "tab button press" to immediately focus available content inside the tab, as opposed to now where you need to "tab your way through" all the tabs before reaching the content of the active one?

Currently, we allow navigation through a set of tabs with both tab / shift tab and the arrow keys.

https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html - a11y docs don't mention use of tab to navigate through items, so maybe we are using a bit of an anti-pattern there. 

It seems like Google Material and IBM Carbon use "tab press" more in the way you have described, and navigation between tab items is only via arrow / home / end keys:

https://material.io/components/tabs
https://www.carbondesignsystem.com/components/tabs/code/

Interestingly, Carbon doesn't require you to hit enter to "activate" the tab, but Material does (a11y refers to these as "manual" and "automatic" modes) 🤔
https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html 
https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html 


To me a good solve would be, allow navigation through tabs only with arrows / home / end, require an "activation" with enter / space, and tab would navigate between the tab nav, and focusable content in the tab container or otherwise the document flow. A nice "plus" feature would be a "selection-mode=manual/automatic" that defaults to "manual" (current behavior) to allow for both selection modes the w3 docs touch on above.

**Comment by janett-baresel** [comment]
> To me a good solve would be, allow navigation through tabs only with arrows / home / end, require an "activation" with enter / space, and tab would navigate between the tab nav, and focusable content in the tab container or otherwise the document flow.

yes this sounds like what i was going for. 

> is there a direct link to a viewable example of this?

We are using the tabs in SV (https://devext.arcgis.com/home/webscene/viewer.html) inside the analysis viewer tool on the right. These tools are a bit complicated because you cannot really use them without a mouse (e.g. measure between 2 points in a scene). Open the measure distance tool and measure something in the scene. Then the tab container shows a UI. If you now play with the tabbing it should do what I described. (Note: when you have not measured anything and enter a tool it automatically switches the focus to the view which "disturbes" the tab order a bit.)

But either way, what you wrote above sounds good. In my opinion the "plus feature" is not strictly needed but I leave this up to you to decide.

**Comment by benelan** [comment]
verified on `beta.93`
